server/lib/shared.py

```diff
@@ -136,3 +136,42 @@ def date_lesser_equal_max(date, max_date):
   if not date:
     return False
   return not max_date or date <= max_date or max_date in date
+
+
+def divide_into_batches(all_items: list, batch_size: int):
+  """Helper function to divide a large list of items into batches of a set
+  batch size. Used to make batched calls to mixer.
+  """
+  for i in range(0, len(all_items), batch_size):
+    yield all_items[i:i + batch_size]
+
+
+def merge_responses(resp_1: dict, resp_2: dict) -> dict:
+  """Merge the response of two calls to the same API into a single response.
+  Requires the two responses to have identical keys. In the case of conflicts,
+  will default to using values from the first given response.
+  """
+  merged_resp = {}
+  for key, val in resp_1.items():
+    if type(val) == dict:
+      if key in resp_2 and type(resp_2[key]) == dict:
+        merged_resp[key] = merge_responses(resp_1[key], resp_2[key])
+      else:
+        # key is not in resp_2, or values conflict, take value from resp_1
+        merged_resp[key] = val
+    elif type(val) == list:
+      if key in resp_2:
+        if type(resp_2[key]) == list:
```

Comment:
I think you also want to keep this if statement combined with 163? because if key is not in resp_2, you also want to set merged_resp[key] to val?

Reply:
I think you're right! It should mirror the structure of the top if. Changed.